### PR TITLE
Remove empty date-parts in issued

### PR DESCRIPTION
**Pull request by dhimmel**
~~WIP: before merging, update manubot branch of dhimmel/csl-styles~~ done in https://github.com/dhimmel/csl-schema/commit/df3d1a3c262eff6da5f3ad1717708640d389f63a

See https://github.com/greenelab/meta-review/pull/101#issuecomment-427953827

Refs https://github.com/greenelab/manubot/issues/66

### CODE REVIEW

**Comment by dhimmel** [comment]
What we used to do:

https://github.com/greenelab/manubot/blob/693fbb7758b5922add30ecaa6e30acb98426a977/manubot/cite/citeproc.py#L88-L95

Note this won't work anymore, since now date-parts is empty rather than being `[None]`.

**Comment by dhimmel** [comment]
As of https://github.com/greenelab/manubot/pull/65/commits/9ad8fa38ef5eeb7a357396316ee00f5efb23020a, `manubot cite --render doi:10.22541/au.149693987.70506124` works and produces:

```
1. The arXiv of the future will not look like the arXiv
Alberto Pepe, Matteo Cantiello, Josh Nicholson
Authorea https://doi.org/gdqbz3
DOI: 10.22541/au.149693987.70506124
```

Note that issued is now an empty object, but does not get entirely removed: `"issued": {},`

File: manubot/cite/citeproc.py [hunk 1]
```diff
@@ -51,20 +51,25 @@ def get_jsonschema_csl_validator():
     import jsonref
     import jsonschema
 
-    url = 'https://github.com/dhimmel/schema/raw/manubot/csl-data.json'
+    url = 'https://github.com/dhimmel/csl-schema/raw/empty-date-parts/csl-data.json'
```
Comment (agitter) [inline manubot/cite/citeproc.py]: What's the difference between these two JSON files?  Does `empty-date-parts` have some short term customization that won't be part of the `manubot` schema?

File: manubot/cite/citeproc.py [hunk 2]
```diff
@@ -51,20 +51,25 @@ def get_jsonschema_csl_validator():
...
     # Use jsonref to workaround https://github.com/Julian/jsonschema/issues/447
     schema = jsonref.load_uri(url, jsonschema=True)
     Validator = jsonschema.validators.validator_for(schema)
     Validator.check_schema(schema)
     return Validator(schema)
 
 
-def remove_jsonschema_errors(instance):
+def remove_jsonschema_errors(instance, recurse_depth=5):
```
Comment (agitter) [inline manubot/cite/citeproc.py]: I trust that it is needed, but is there an easy-to-describe case where calling `remove_jsonschema_errors` once is insufficient but calling it multiple times recursively works?  I take it empty date-parts is such a case?

File: tests/csl-json/excess-date-parts-csl/raw.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+[
```
Comment (agitter) [inline tests/csl-json/excess-date-parts-csl/raw.json]: What does `excess-date-parts` refer to in this test case?  I'm not sure what is being tested.

File: manubot/cite/citeproc.py (hunk 1 above)
Comment (dhimmel) [reply manubot/cite/citeproc.py]: Before merging we will revert this PR is using the manubot CSL branch. However, I will add this https://github.com/dhimmel/csl-schema/commit/b0128abc7f7dd78bf140607f339273966fcc1bba from `empty-date-parts` onto that branch.

Basically, I needed to test this PR with the new CSL JSON Schema.

File: tests/csl-json/excess-date-parts-csl/raw.json (hunk 3 above)
Comment (dhimmel) [reply tests/csl-json/excess-date-parts-csl/raw.json]: Fixed in https://github.com/greenelab/manubot/pull/65/commits/4428dadc07b310861d943b63be9a59d45f62d2cf

Comment (dhimmel) [reply tests/csl-json/excess-date-parts-csl/raw.json]: Fixed in https://github.com/greenelab/manubot/pull/65/commits/dfda71cb9ee0e8bffa7bae09e07e6791c7c14707

File: manubot/cite/citeproc.py (hunk 2 above)
Comment (agitter) [reply manubot/cite/citeproc.py]: Thanks for the detailed explanation.

**Comment by dhimmel** [comment]
I forgot to change the JSON Schema URL back, so I amended the merged commit in master. The new commit that this has been merged into is https://github.com/greenelab/manubot/commit/4e6a0f6d28220264c0d7892c732cb68c3e97c07a

**Pull request merged**